courses/task_four/main.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+)
+
+// averageArray that calculate average value
+func averageArray(numbers [6]int) float64 {
+	var sum int = 0
+	for _, value := range numbers {
+		sum += value
+	}
+	return float64(sum) / float64(len(numbers))
+}
+
+// maxStringValue fthat returns the longest word
+func maxStringValue(s []string) (maxVal string) {
+	for _, value := range s {
+		if len(value) > len(maxVal) {
+			maxVal = value
+		}
+	}
+	return
+}
+
+// reverseSlice that returns the reversed copy
+func reverseSlice(s []int64) (reversed []int64) {
+	for i := 0; i < len(s); i++ {
+		reversed = append(reversed, s[len(s)-1-i])
```

Comment:
This func is safe, but it will create a slice with lot of empty spaces, up to 2X times it requires.
```suggestion
reversed  = make([]int64, 0, len(s))
for i := 0; i < len(s); i++ {
    reversed = append(reversed, s[len(s)-1-i])
}
```
https://gist.github.com/wshaman/af385cdd97e8c0d10f4eb86b3dedf20c

Reply:
!ok... excellent explanation